prosessering/src/main/java/no/nav/familie/prosessering/internal/TaskWorker.kt

```diff
@@ -47,83 +45,57 @@ class TaskWorker(private val taskRepository: TaskRepository, taskStepTyper: List
     }
 
 
-    @Async("taskExecutor")
-    fun doTaskStep(taskId: Long?) {
-        requireNotNull(taskId, { "taskId kan ikke være null" })
-        doActualWork(taskId)
-    }
-
-    // For Unit testing
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
     fun doActualWork(taskId: Long) {
-        val startTidspunkt = System.currentTimeMillis()
-        var maxAntallFeil = 0
-        var task = taskRepository.findByIdOrNull(taskId) ?: error("Kunne ikke finne task med id $taskId")
-
-        initLogContext(task)
-        try {
-
-            secureLog.trace("Behandler task='{}'", task)
-            task.behandler()
-            task = taskRepository.saveAndFlush(task)
-
-            // finn tasktype
-            val taskStep = finnTaskStep(task.taskStepType)
-            maxAntallFeil = finnMaxAntallFeil(task.taskStepType)
-
-            // execute
-            execute(taskStep, task)
-
-            task.ferdigstill()
-            secureLog.trace("Ferdigstiller task='{}'", task)
-            taskRepository.saveAndFlush(task)
-            taskRepository.flush()
-            secureLog.info("Fullført kjøring av task '{}', kjøretid={} ms",
-                           task,
-                           System.currentTimeMillis() - startTidspunkt)
-        } catch (e: Exception) {
-            task.feilet(TaskFeil(task, e), maxAntallFeil)
-            // lager metrikker på tasks som har feilet max antall ganger.
-            if (task.status == Status.FEILET) {
-                finnFeilteller(task.taskStepType).increment()
-            }
-            secureLog.warn("Fullført kjøring av task '{}', kjøretid={} ms, feilmelding='{}'",
-                           task,
-                           System.currentTimeMillis() - startTidspunkt,
-                           e)
-            task.triggerTid = task.triggerTid?.plusSeconds(finnTriggerTidVedFeil(task.taskStepType))
-            taskRepository.save(task)
-            secureLog.info("Feilhåndtering lagret ok {}", task)
-        } finally {
-            clearLogContext()
+
+        val taskOptional = taskRepository.findById(taskId)
+
+        if (taskOptional.isEmpty) {
+            return
         }
-    }
 
-    @Transactional(propagation = Propagation.REQUIRES_NEW)
-    fun execute(taskStep: AsyncTaskStep, task: Task) {
+        val task = taskOptional.get()
+        if (task.status != Status.PLUKKET) {
+            return // en annen pod har startet behandling
+        }
```

Comment:
Dette bør vel ikke skje med leader-election uansett, men greit med litt bukseseler her :)

Reply:
Bør ikke skje med leader election, men det ser ut til at vi klarte å konfigurere ef-mottak feil. Tror den ene poden kan ha fått leader election = true og den andre null. Da kjørrte leader gankse usikret i tilleg til at den andre også kjørte.

> Virker fornuftig dette. Skjønte ikke helt launch opplegget i testene, men stoler på at det er testet godt nok.
> 
> Er låsen mot db fjernet? Hva skjer f.eks. hvis 2 poder prøver å lagre "plukker" status samtidig?
Låsen mot db er forbedret, så det skal være mindre sjanse (forhåpentligvis umulig) for konflikter